src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTestTransformer.java

```diff
@@ -256,6 +215,117 @@ private static ModelTypeTable createTypeTable(String packageName) {
     return testCaseViews;
   }
 
+  private MockServiceImplFileView createMockServiceImplView(GapicInterfaceContext context) {
+    addUnitTestImports(context.getModelTypeTable());
+
+    Interface apiInterface = context.getInterface();
+    SurfaceNamer namer = context.getNamer();
+    String outputPath =
+        PhpGapicCodePathMapper.newBuilder()
+            .setPrefix("tests/unit")
+            .build()
+            .getOutputPath(context.getInterface(), context.getProductConfig());
+    String name = namer.getMockGrpcServiceImplName(context.getInterface());
+    String grpcClassName =
+        context
+            .getModelTypeTable()
+            .getAndSaveNicknameFor(namer.getGrpcClientTypeName(apiInterface));
+
+    MockServiceImplFileView.Builder mockServiceImplFile = MockServiceImplFileView.newBuilder();
+
+    mockServiceImplFile.serviceImpl(
+        MockServiceImplView.newBuilder()
+            .name(name)
+            .grpcClassName(grpcClassName)
+            .grpcMethods(new ArrayList<MockGrpcMethodView>())
+            .build());
+
+    mockServiceImplFile.outputPath(namer.getSourceFilePath(outputPath, name));
+    mockServiceImplFile.templateFileName(MOCK_SERVICE_IMPL_TEMPLATE_FILE);
+
+    ImportSectionView importSection =
+        importSectionTransformer.generateImportSection(context.getModelTypeTable().getImports());
+    PhpSurfaceNamer testPackageNamer =
+        new PhpSurfaceNamer(namer.getTestPackageName(SurfaceNamer.TestKind.UNIT));
```

Comment:
You could instead do `namer.cloneWithPackageName(...)`

Reply:
Cool! Done